tests/unit/dataset/loader/test_single_turn.py

```
@@ -15,6 +15,31 @@
     """Basic functionality tests for SingleTurn model."""
 
+    def test_create_with_input_alias_for_text(self):
+        """JSONL may use ``input``; it must populate ``text`` for modality validation."""
+        data = SingleTurn.model_validate(
+            {"type": "single_turn", "input": "What is deep learning?"}
+        )
+        assert data.text == "What is deep learning?"
+
+    def test_input_and_text_conflict_raises(self):
+        """Providing 'input' and 'text' with different values must raise a ValidationError."""
+        with pytest.raises(ValueError, match="input.*text"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)